src/FirebaseAuth.js

```diff
@@ -222,24 +222,27 @@
      */
     _routerMethodOnAuthPromise: function(rejectIfAuthDataIsNull) {
       var ref = this._ref;
-      var deferred = this._q.defer();
 
-      function callback(authData) {
-        if (authData !== null) {
-          deferred.resolve(authData);
-        } else if (rejectIfAuthDataIsNull) {
-          deferred.reject("AUTH_REQUIRED");
-        } else {
-          deferred.resolve(null);
+      return this._utils.promise(function(resolve,reject){
+        function callback(authData) {
+          // Turn off this onAuth() callback since we just needed to get the authentication data once.
+          ref.offAuth(callback);
+
+          if (authData !== null) {
+            resolve(authData);
+            return;
```

Comment:
I think you might be conflating these methods with `$q.resolve` and `$q.reject`, which return promises. These methods behave like `deferred.resolve` and `deferred.reject` which do not have specified return values. The original code provides chain-able promises.

See the [$q constructor documentation](https://docs.angularjs.org/api/ng/service/$q), they aren't returning there at all. I would think it's bad practice to return the result of a function whose return value is guaranteed by documentation / specification (they may be leaving it's return value unspecified intentionally to allow for future changes).